test/oauth-server/service/token.go

```diff
@@ -214,5 +289,8 @@ func (requestHandler *RequestHandler) processResponse(w http.ResponseWriter, tok
 		"refresh_token": "refresh-token",
 	}
 
-	jsonResponseWriter(w, resp)
+	if err = jsonResponseWriter(w, resp); err != nil {
+		writeError(w, err, http.StatusBadRequest)
```

Comment:
when w.Write fails you can only log error. it is not recommended to call twice w.WriteHeader